backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -152,16 +156,225 @@ const testGetUsername = () => {
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have weak newPassword', async () => {
+			const data = { oldPassword: testUser.password, newPassword: 'abc' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the update data have a newPassword which is the same as the old one', async () => {
+			const data = { oldPassword: testUser.password, newPassword: testUser.password };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should update the profile if the user is logged in', async () => {
+			const data = { firstName: 'newName' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.put('/v5/user/').send(data).expect(200);
+		});
+
+		test('should update the profile and change password if the user is logged in', async () => {
+			const data = { firstName: 'newName', oldPassword: testUser.password, newPassword: 'Passport123!' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.put('/v5/user/').send(data).expect(200);
+			await testSession.put('/v5/user/').send({ oldPassword: 'Passport123!', newPassword: testUser.password });
+		});
+	});
+};
+
+const testGenerateApiKey = () => {
+	describe('Generate and assign Api key to the logged in user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const res = await agent.post('/v5/user/key').expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user has a session via an API key', async () => {
+			const res = await agent.post(`/v5/user/key?key=${testUser.apiKey}`)
+				.expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should create and return a new Api key if the user is logged in', async () => {
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.post('/v5/user/key').expect(200);
+			const userProfileRes = await testSession.get('/v5/user');
+			expect(res.body).toEqual({ apiKey: userProfileRes.body.apiKey });
```

Comment:
Can we verify the api key works?

Reply:
Also add a test where there's already an existing api key and generating a new one means the old one should stop working

Reply:
Be good to call logout after this test to we're not contaminating the next test it runs (applies to other places but i'll stop pointing it out)